src/statemachine/statemachine_coroutine.js

```diff
@@ -0,0 +1,225 @@
+pc.extend(pc, (function () {
+
+    var coroutines = [];
+
+    function noop() {
+    }
+
+    pc.ComponentSystem.on('update', function (dt) {
```

Comment:
I don't like this dependency on the page-global component system update.

It would be preferable to hook up and define an update function here, and then hook up the update in the application initialization.

Ideally, we should steer clear of any "Singleton" type behaviour, (I know we have it already in these global update fns, but I'd like to get rid of it). It means multiple applications on the same page can interfere with each other. Possibly we need some sort of StatemachineManager instance which has the update fn and holds the instances. This would be created and updated in the Application.